src/platform/Linux/NetworkCommissioningDriver.cpp

```diff
@@ -0,0 +1,269 @@
+                            memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                        Status::kNetworkNotFound);
+
+    mStagingNetwork.ssidLen = 0;
+    return Status::kSuccess;
+}
+
+void LinuxWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback)
+{
+    CHIP_ERROR err          = CHIP_NO_ERROR;
+    Status networkingStatus = Status::kSuccess;
+
+    // We only support one network, so reorder is actually no-op.
+    VerifyOrExit(networkId.size() == mStagingNetwork.ssidLen &&
+                     memcmp(networkId.data(), mStagingNetwork.ssid, mStagingNetwork.ssidLen) == 0,
+                 networkingStatus = Status::kNetworkNotFound);
+
+    ChipLogProgress(NetworkProvisioning, "LinuxNetworkCommissioningDelegate: SSID: %s", networkId.data());
+
+    err = ConnectivityMgrImpl().ProvisionWiFiNetwork(reinterpret_cast<const char *>(mStagingNetwork.ssid),
+                                                     reinterpret_cast<const char *>(mStagingNetwork.credentials));
+
+exit:
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(NetworkProvisioning, "Failed to connect to WiFi network: %s", chip::ErrorStr(err));
+        networkingStatus = Status::kUnknownError;
+    }
+
+    callback->OnResult(networkingStatus, CharSpan(), 0);
+}
+
+void LinuxWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * callback)
+{
+    CHIP_ERROR err = DeviceLayer::ConnectivityMgrImpl().StartWiFiScan(ssid, callback);
+    if (err != CHIP_NO_ERROR)
+    {
+        callback->OnFinished(Status::kUnknownError, CharSpan(), nullptr);
+    }
+}
+
+size_t LinuxWiFiDriver::WiFiNetworkIterator::Count()
+{
+    return driver->mStagingNetwork.ssidLen == 0 ? 0 : 1;
+}
+
+bool LinuxWiFiDriver::WiFiNetworkIterator::Next(Network & item)
+{
+    if (exhausted || driver->mStagingNetwork.ssidLen == 0)
+    {
+        return false;
+    }
+    memcpy(item.networkID, driver->mStagingNetwork.ssid, driver->mStagingNetwork.ssidLen);
+    item.networkIDLen = driver->mStagingNetwork.ssidLen;
+    item.connected    = false;
+    exhausted         = true;
+    return true;
+}
+
+#endif
+
+CHIP_ERROR LinuxThreadDriver::CommitConfiguration()
+{
+    mRunningNetwork = mStagingNetwork;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR LinuxThreadDriver::RevertConfiguration()
+{
+    mStagingNetwork = mRunningNetwork;
+    return CHIP_NO_ERROR;
+}
+
+Status LinuxThreadDriver::AddOrUpdateNetwork(ByteSpan operationalDataset)
+{
+    uint8_t extpanid[kSizeExtendedPanId];
+    uint8_t newExtpanid[kSizeExtendedPanId];
+    Thread::OperationalDataset newDataset;
+
+    newDataset.Init(operationalDataset);
+    VerifyOrReturnError(newDataset.IsCommissioned(), Status::kOutOfRange);
+
+    VerifyOrReturnError(!mStagingNetwork.IsCommissioned() || memcmp(extpanid, newExtpanid, kSizeExtendedPanId) == 0,
+                        Status::kBoundsExceeded);
+
+    mStagingNetwork = newDataset;
+    return Status::kSuccess;
+}
```

Comment:
We should have Thread/Wifi code conditional on config supported. Otherwise platforms will cut'and'paste this whole file and it will get confusing. We need to find a clean way to make the cluster not depend on the entire presence of both Thread/Wifi code stubs

Reply:
Splited this file into two files.